src/manifest.c

```diff
@@ -776,24 +779,49 @@ static int write_manifest_plain(struct manifest *manifest)
 	}
 
 	list = g_list_first(manifest->manifests);
+	string_or_die(&manifest_tempdir, "./XXXXXX");
```

Comment:
Can this tempdir be created in the state directory instead? So, by default, the template would be /var/lib/update/XXXXXX